teek/_widgets/treeview.py

```
@@ -0,0 +1,545 @@
+                                                  'selection')
+
+    @make_thread_safe
+    def select(self):
+        """
+        Add row to current selection.
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'add', self)
+
+    @make_thread_safe
+    def deselect(self):
+        """
+        Remove row from current selection.
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'remove', self)
+
+    @make_thread_safe
+    def move(self, to_pos):
+        """
+        Move row to other position.
+        """
+        self._treeview._call(
+            None, self._treeview, 'move', self, self._root, to_pos
+        )
+
+    def to_tcl(self):
+        # Represented by name
+        return self._name
+
+
+class TreeviewColumnList(MutableSequence):
+    """
+    List containing all columns of a treeview.
+
+    Note, that instances of :class:`TreeviewColumnList` must not be created
+    manually. It is accessible through the `.columns` attribute.
+    """
+    def __init__(self, treeview, init_columns=[]):
+        super().__init__()
+
+        # Default column 0
+        first_column = TreeviewColumn('#0')
+        first_column._assign(treeview)
+
+        self._data = [first_column]
+        self._treeview = treeview
+
+        for column in init_columns:
+            self.append(column)
+
+    def __repr__(self):
+        return '%s(columns=%d)' % (
+            self.__class__.__name__,
+            len(self)
+        )
+
+    @make_thread_safe
+    def __setitem__(self, index, column):
+        if index == 0 or index <= -len(self):
+            raise KeyError('cannot set column #0')
+
+        if not isinstance(column, TreeviewColumn):
+            column = TreeviewColumn(text=column)
+
+        self._data[index] = column
+        self._assign()
+
+    @make_thread_safe
+    def __delitem__(self, index):
+        if index == 0:
```

This won't detect `del columnlist[0:1]` and `del columnlist[len(treeview) - 1]`. The slice thing is quite difficult to check, so you could just raise an error if someone attempts to delete a slice of the column list (and mention in the docs that deleting a slice of a column list isn't supported). It's easy to check for the `len-1` thing.